include/deal.II/numerics/vector_tools_interpolate.templates.h

```diff
@@ -526,6 +526,8 @@ namespace VectorTools
     VectorType &                                               vec,
     const ComponentMask &                                      component_mask)
   {
+    AssertDimension(dof.get_fe_collection().n_components(),
+                    function.n_components);
```

Comment:
Are we sure we still need this with the current code? I tried this and got an exception a few lines above, where this function ends up eventually, https://github.com/dealii/dealii/blob/5df4e622f788a831734f48871e25f22115b10b23/include/deal.II/numerics/vector_tools_interpolate.templates.h#L485-L486

I believe this got implicitly fixed once we merged `hp::DoFHandler` and `DoFHandler` in 2020, at least I could not find a way to make this reproduce.

Reply:
It also doesn't hurt to have it here, doesn't it?

Reply:
I couldn't convince myself that there aren't call paths where we don't get the exception. So I just added it, under the same premise as @tjhei mentions.